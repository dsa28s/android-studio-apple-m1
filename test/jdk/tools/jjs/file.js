/*
 * This is the test JavaScript program used in jjs-fileTest.sh
 */

// good old 'hello world'!
print('hello');

// basic number manipulation
var v = 2 + 5;
v *= 5;
v.doubleValue();
v = v + " is the value";
if (v != 0) {
    print('yes v != 0');
}

// basic java access
java.lang.System.out.println('hello world from script');

// basic stream manipulation
var al = new java.util.ArrayList();
al.add("hello");
al.add("world");
// script functions for lambas
al.stream().map(function(s) s.toUpperCase()).forEach(print);

// interface implementation
new java.lang.Runnable() {
    run: function() {
        print('I am runnable');
    }
}.run();

// java class extension
var MyList = Java.extend(java.util.ArrayList);
var m = new MyList() {
    size: function() {
        print("size called");
        // call super.size()
        return Java.super(m).size();
    }
};

print("is m an ArrayList? " + (m instanceof java.util.ArrayList));
m.add("hello");
m.add("world");
print(m.size());
