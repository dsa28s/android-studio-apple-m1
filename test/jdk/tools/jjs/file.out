hello
yes v != 0
hello world from script
HELLO
WORLD
I am runnable
is m an ArrayList? true
size called
2
